tilescopethree/strategies/decomposition_strategies/factor.py

```diff
@@ -1,12 +1,40 @@
 from itertools import chain
 
 from comb_spec_searcher import Rule
-from permuta.misc import UnionFind
-from tilings import Tiling
-from tilings.misc import union_reduce
+from tilings.algorithms import (Factor, FactorWithInterleaving,
+                                FactorWithMonotoneInterleaving)
+
+
+def general_factor(tiling, factor_class, workable=True, union=False):
+    """
+    Iterator of factor strategy.
+    """
+    factor = factor_class(tiling)
+    if factor.factorable():
+        yield factor.rule(workable=workable)
+        if union:
+            yield from factor.all_union_rules(workable=workable)
```

Comment:
For union workable shoud be false by default

Reply:
I changed that. Also, if a kwargs `workable` is passed to the general factor iterator, it will only affect the irreducible factorisation rule. Do you agree with this behaviour @christianbean ?